lib/pacemaker/pcmk_sched_bundle.c

```diff
@@ -555,18 +555,28 @@ pcmk__bundle_with_colocations(const pe_resource_t *rsc,
     }
 }
 
+/*!
+ * \internal
+ * \brief Return action flags for a given bundle resource action
+ *
+ * \param[in,out] action  Bundle resource action to get flags for
+ * \param[in]     node    If not NULL, limit effects to this node
+ *
+ * \return Flags appropriate to \p action on \p node
+ */
 enum pe_action_flags
 pcmk__bundle_action_flags(pe_action_t *action, const pe_node_t *node)
 {
     GList *containers = NULL;
     enum pe_action_flags flags = 0;
-    pe_resource_t *bundled_resource = pe__bundled_resource(action->rsc);
+    pe_resource_t *bundled_resource = NULL;
 
-    if (bundled_resource != NULL) {
-        enum action_tasks task = get_complex_task(bundled_resource,
-                                                  action->task);
+    CRM_ASSERT((action != NULL) && (action->rsc != NULL)
+               && (action->rsc->variant == pe_container));
 
-        switch(task) {
+    bundled_resource = pe__bundled_resource(action->rsc);
+    if (bundled_resource != NULL) {
+        switch (get_complex_task(bundled_resource, action->task)) {
```

Comment:
It's not immediately obvious why we use `bundled_resource->children` for certain tasks if `bundled_resource` exists, and the list of the bundle's containers otherwise. It's probably not hard to figure out but one or two brief comments might be warranted.

Reply:
makes sense